NotificationService/NotificationService.BusinessLibrary/Providers/MSGraphNotificationProvider.cs

```diff
@@ -192,12 +192,27 @@ private async Task ProcessEntitiesIndividually(string applicationName, IList<Ema
                     }
 
                     EmailMessagePayload payLoad = new EmailMessagePayload(message) { SaveToSentItems = saveToSent };
-                    var isSuccess = await this.msGraphProvider.SendEmailNotification(emailAccountUsed.Item1, payLoad, item.NotificationId).ConfigureAwait(false);
-
-                    item.Status = isSuccess ? NotificationItemStatus.Sent : (item.TryCount <= this.maxTryCount ? NotificationItemStatus.Retrying : NotificationItemStatus.Failed);
+                    var response = await this.msGraphProvider.SendEmailNotification(emailAccountUsed.Item1, payLoad, item.NotificationId).ConfigureAwait(false);
+                    if (response.Status)
+                    {
+                        item.Status = NotificationItemStatus.Sent;
+                    }
+                    else if (item.TryCount <= this.maxTryCount && (response.StatusCode == HttpStatusCode.TooManyRequests || response.StatusCode == HttpStatusCode.RequestTimeout))
+                    {
+                        item.Status = NotificationItemStatus.Retrying;
+                        item.ErrorMessage = response.Result;
+                        _ = this.LogMailBoxExhaustedTelemetry(response.Result, item.NotificationId, item.EmailAccountUsed, false, traceProps);
```

Comment:
why MailBoxExhaused for too many requests?

Reply:
As i could see the mailbox limit exhausted was added under toomanyrequest in processbatch method. I used the same in processindividually and in meeting invite processing as well. Becuase the same mailbox exhausted issue could appear in these case as well.